ci-tests/test-spike

```diff
@@ -6,7 +6,7 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 mkdir build
 cd build
 mkdir install
-CFLAGS="-Werror" $DIR/../configure --prefix=`pwd`/install
+CFLAGS="-Werror -Wignored-qualifiers -Wunused-function -Wunused-parameter -Wunused-variable" $DIR/../configure --prefix=`pwd`/install
```

Comment:
Should we add `-Wnon-virtual-dtor` to this?

Reply:
Yep, I had excluded it because I thought our CI's compiler didn't support it, but that was my error.  I needed to add it to CXXFLAGS, not CFLAGS.